### ISSUE TITLE
[qualys_vmdr.user_activity] Handle response header / footer

### ISSUE THREAD

**Issue by andrewkroh** [elastic/integrations#10183]
The CSV response from the activity_logs API includes a content that is not CSV (e.g. `----BEGIN_RESPONSE_BODY_CSV`, `----END_RESPONSE_BODY_CSV`). It can also include a footer that contains a URL that can be used to continue pagination. This behavior is not described in the [documentation](https://cdn2.qualys.com/docs/qualys-api-vmpc-user-guide.pdf).

### Non-truncated response body

```
----BEGIN_RESPONSE_BODY_CSV
"Date","Action","Module","Details","User Name","User Role","User IP"
"2024-06-18T12:45:24Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
----END_RESPONSE_BODY_CSV
```

### Truncated response (includes headers as logged by curl)

```
< HTTP/2 200 
< date: Tue, 18 Jun 2024 12:47:48 GMT
< content-type: text/csv;charset=UTF-8
< x-content-type-options: nosniff
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< x-powered-by: Qualys:XXX
< x-ratelimit-limit: 300
< x-ratelimit-window-sec: 3600
< x-concurrency-limit-limit: 2
< x-concurrency-limit-running: 0
< x-ratelimit-towait-sec: 0
< x-ratelimit-remaining: 295
< content-security-policy: default-src 'self' *.qualys.com *.qualys.ca *.qualys.eu *.qualys.it *.qualys.in *.qualys.ae *.qualys.co.uk *.qualys.com.au *.qualysksa.com *.qualys.sg  *.qualys.us;
< strict-transport-security: max-age=31536000; includeSubDomains
< x-xss-protection: 1; mode=block
< cf-cache-status: DYNAMIC
< server: cloudflare
< cf-ray: XXX
< 

----BEGIN_RESPONSE_BODY_CSV
"Date","Action","Module","Details","User Name","User Role","User IP"
"2024-06-18T12:45:24Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:58Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:38Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
----END_RESPONSE_BODY_CSV
----BEGIN_RESPONSE_FOOTER_CSV
WARNING
"CODE","TEXT","URL"
"1980","3 record limit exceeded. Use URL to get next batch of results.","https://foo.apps.qualys.com/api/2.0/fo/activity_log/?action=list&since_datetime=2024-06-16T22%3a00%3a00Z&truncation_limit=3&id_max=1425858279"
----END_RESPONSE_FOOTER_CSV
```

**Comment by efd6** [comment]
The best I can come up with for this is
```
mito src.cel
! stderr .
cmp stdout want.txt

-- src.cel --
[
b"""
----BEGIN_RESPONSE_BODY_CSV
"Date","Action","Module","Details","User Name","User Role","User IP"
"2024-06-18T12:45:24Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:58Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:38Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
----END_RESPONSE_BODY_CSV
----BEGIN_RESPONSE_FOOTER_CSV
WARNING
"CODE","TEXT","URL"
"1980","3 record limit exceeded. Use URL to get next batch of results.","https://foo.apps.qualys.com/api/2.0/fo/activity_log/?action=list&since_datetime=2024-06-16T22%3a00%3a00Z&truncation_limit=3&id_max=1425858279"
----END_RESPONSE_FOOTER_CSV
""",
b"""
"Date","Action","Module","Details","User Name","User Role","User IP"
"2024-06-18T12:45:24Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:58Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"2024-06-18T12:44:38Z","request","auth","API: /api/2.0/fo/activity_log/index.php","johndoe","Reader","175.16.199.1"
"""
].map(b, string(b).as(text, {
	"csv": (
	  text.contains_substr("----BEGIN_RESPONSE_BODY_CSV") ?
	    bytes(text.substring(
	      text.index("----BEGIN_RESPONSE_BODY_CSV")+size("----BEGIN_RESPONSE_BODY_CSV\n"),
	      text.index("----END_RESPONSE_BODY_CSV")
	    ))
	  :
	    b
	  ).mime("text/csv; header=present"),
	?"next": text.contains_substr("----BEGIN_RESPONSE_FOOTER_CSV") ?
	  optional.of(bytes(text.substring(
	    text.index("----BEGIN_RESPONSE_FOOTER_CSV")+size("----BEGIN_RESPONSE_FOOTER_CSV\n"),
	    text.index("----END_RESPONSE_FOOTER_CSV")
	  ).trim_prefix("WARNING\n")).mime("text/csv; header=present"))
	:
	  optional.none(),
}))
-- want.txt --
[
	{
		"csv": [
			{
				"Action": "request",
				"Date": "2024-06-18T12:45:24Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			},
			{
				"Action": "request",
				"Date": "2024-06-18T12:44:58Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			},
			{
				"Action": "request",
				"Date": "2024-06-18T12:44:38Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			}
		],
		"next": [
			{
				"CODE": "1980",
				"TEXT": "3 record limit exceeded. Use URL to get next batch of results.",
				"URL": "https://foo.apps.qualys.com/api/2.0/fo/activity_log/?action=list&since_datetime=2024-06-16T22%3a00%3a00Z&truncation_limit=3&id_max=1425858279"
			}
		]
	},
	{
		"csv": [
			{
				"Action": "request",
				"Date": "2024-06-18T12:45:24Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			},
			{
				"Action": "request",
				"Date": "2024-06-18T12:44:58Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			},
			{
				"Action": "request",
				"Date": "2024-06-18T12:44:38Z",
				"Details": "API: /api/2.0/fo/activity_log/index.php",
				"Module": "auth",
				"User IP": "175.16.199.1",
				"User Name": "johndoe",
				"User Role": "Reader"
			}
		]
	}
]
```
I don't like the memory behaviour here. We don't have bytes search/substring currently, which would make it better.

**Comment by andrewkroh** [comment]
This has been addressed in version 4.1.1 of the qualys vmdr integration.
